exercises/practice/phone-number/.meta/Example.roc

```diff
@@ -0,0 +1,30 @@
+module [clean]
+
+clean : Str -> Result Str [InvalidNumber]
+clean = \phoneNumber ->
+    digits =
+        phoneNumber
+        |> Str.toUtf8
+        |> List.keepIf \c -> c >= '0' && c <= '9'
+
+    numDigits = List.len digits
+    if numDigits == 10 then
+        digits |> checkNumber
+    else if numDigits == 11 then
+        when digits is
+            ['1', .. as rest] -> rest |> checkNumber
+            _ -> Err InvalidNumber
+    else
+        Err InvalidNumber
+
+checkNumber : List U8 -> Result Str [InvalidNumber]
+checkNumber = \digits ->
+    areaStart = digits |> List.get 0 |> Result.withDefault '0'
+    exchangeStart = digits |> List.get 3 |> Result.withDefault '0'
+    if areaStart == '0' || areaStart == '1' || exchangeStart == '0' || exchangeStart == '1' then
+        Err InvalidNumber
+        else
```

Comment:
```suggestion
    when digits is
        [areaStart, _, _, exchangeStart, _, _, _, _, _, _] if areaStart != '0' ... -> 
```
You could do something like this here to remove the need for handling the results. The trailing wildcards could also be replaced with `..` for brevity since the length has already been validated in `clean`.

The current style is definitely also a good approach, but I would suggest adjusting it like this to make the behavior a bit more explicit:
```roc
areaStart = digits |> List.get 0 |> Result.mapErr? InvalidNumber
```

Reply:
Oh great idea, thanks, I'll do that!

Reply:
Actually I think the second snippet would need to be this instead:
```roc
areaStart = digits |> List.get 0 |> Result.mapErr? \_ -> InvalidNumber
```
But haven't had the chance to actually verify 😅

Reply:
Nice suggestion! However, I've already refactored the code using your pattern-matching suggestion, and I quite like it. 👍